igvc_platform/src/tests/test_sim_color_detector.cpp

```diff
@@ -0,0 +1,200 @@
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+#include <mocking_utils/mock_subscriber.h>
+#include <sensor_msgs/Image.h>
+#include <sensor_msgs/CameraInfo.h>
+
+#include <cv_bridge/cv_bridge.h>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/highgui/highgui.hpp>
+
+class TestSimColorDetector : public testing::Test
+{
+public:
+  TestSimColorDetector()
+    : mock_image_pub(handle.advertise<sensor_msgs::Image>("/cam/center/raw/image", 1))
+    , mock_info_pub(handle.advertise<sensor_msgs::CameraInfo>("/cam/center/raw/camera_info", 1))
```

Comment:
This only tests a single hardcoded camera. It would nice to test all the cameras using the parameters given in the test launch file. This would be quite similar to the implementation in the sim color detector node.